sdk/burdock/sql/private_rewriter.py

```diff
@@ -148,11 +150,16 @@ def exact_aggregates(self, query):
         select = Select(None, select)
 
         subquery = Query(child_scope.select(), query.source, query.where, query.agg, query.having, None)
-        subquery = self.per_key_random(subquery)
-        subquery = [AliasedRelation(subquery, "per_key_random")]
+        if self.options.reservoir_sample:
+            subquery = self.per_key_random(subquery)
+            subquery = [AliasedRelation(subquery, "per_key_random")]
 
-        filtered = Where(BooleanCompare(Column("per_key_random.row_num"), "<=", Literal(str(self.max_contrib), self.max_contrib)))
-        return Query(select, From(subquery), filtered, query.agg, None, None)
+            filtered = Where(BooleanCompare(Column("per_key_random.row_num"), "<=", Literal(str(self.options.max_contrib), self.options.max_contrib)))
```

Comment:
Can we disable this filter in subquery through options?

Reply:
Yes, that will automatically be skipped if you do private.options.reservoir_sample = False